kernel/rtlil.h

```diff
@@ -728,6 +728,7 @@ struct RTLIL::SigBit
 	SigBit(const RTLIL::SigBit &sigbit) = default;
 	RTLIL::SigBit &operator =(const RTLIL::SigBit &other) = default;
 
+	std::string str() const;
```

Comment:
The canonical form of creating a string representation of a signal is `log_signal()`. (Yes, `log_signal()` would add a blank between the wire name and the index. If that's a problem then the pass should have an application specific function for turning a SigBit into a string. No global `str()` function with a format different from the canonical `log_signal()` format should be needed.)

Reply:
That's fair.  Actually, I think it's preferable to go that route anyway, as a space character is the only character I'm 100% positive will always be disallowed in wire names, and is visually much nicer as a separator character than `[`, `]`, and `#`.

Reply:
> as a space character is the only character I'm 100% positive will always be disallowed in wire names, and is visually much nicer as a separator character than `[`, `]`, and `#`.

To be clear, all characters except control and space are explicitly *allowed* in wire names--I spent some time clarifying this just recently.

Reply:
I see, thanks for the clarification.  I updated the PR to ensure the wire name is always surrounded by spaces and is described in the recovery regex as `[^ ]*` instead of `.*`.